pkg/core/interop_neo.go

```diff
@@ -119,22 +119,19 @@ func contractUpdate(ic *interop.Context) error {
 	if contract == nil {
 		return errors.New("contract doesn't exist")
 	}
-	script := ic.VM.Estack().Pop().Bytes()
-	if len(script) > MaxContractScriptSize {
-		return errors.New("the script is too big")
+	script := ic.VM.Estack().Pop().BytesOrNil()
+	if l := len(script); script != nil && (l == 0 || l > MaxContractScriptSize) {
+		return fmt.Errorf("invalid script size: %d (max %d)", l, MaxContractScriptSize)
 	}
-	manifestBytes := ic.VM.Estack().Pop().Bytes()
-	if len(manifestBytes) > manifest.MaxManifestSize {
-		return errors.New("manifest is too big")
+	manifestBytes := ic.VM.Estack().Pop().BytesOrNil()
+	if l := len(manifestBytes); manifestBytes != nil && (l == 0 || l > manifest.MaxManifestSize) {
+		return fmt.Errorf("invalid manifest size: %d (max %d)", l, manifest.MaxManifestSize)
```

Comment:
Then we also have to return an error after validation in case if both script and manifest are nil. See [the reference implementation](https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/ApplicationEngine.Contract.cs#L67).

Reply:
Fixed.